sources/advanced/Command.ts

```diff
@@ -260,31 +270,121 @@ export abstract class Command<Context extends BaseContext = BaseContext> {
+                command.addOption({names: optNames, arity: length, hidden, allowBinding: false});
             });
 
             this.registerTransformer(prototype, (state, command) => {
-                for (const {name, value} of state.options) {
-                    if (optNames.includes(name)) {
-                        // @ts-ignore: The property is meant to have been defined by the child class
-                        command[propertyName] = command[propertyName] || [];
-                        // @ts-ignore: The property is meant to have been defined by the child class
-                        command[propertyName].push(value);
-                    }
-                }
+                const values = state.options
+                    .filter(({name}) => optNames.includes(name))
+                    .map(({value}) => value);
+
+                while (values.length > 0)
+                    // @ts-ignore: The property is meant to have been defined by the child class
+                    command[propertyName] = values.splice(0, length);
             });
         };
     }
 
+    /**
+     * Register an option modifier that causes the attached listener to push its arguments into the array represented in the property.
+     */
+    static Array(): PropertyDecorator;
+
+    /**
+     * Register a listener that looks for an option and its followup argument. When Clipanion detects that this argument is present, the value will be pushed into the array represented in the property.
+     *
+     * @deprecated `Command.Array` should be composed together as a modifier with other option decorators.
+     */
+    static Array(descriptor: string, opts?: {hidden?: boolean}): PropertyDecorator;
+
+    static Array(descriptor?: string, {hidden = false}: {hidden?: boolean} = {}) {
+        return <Context extends BaseContext>(prototype: Command<Context>, propertyName: string) => {
+            // The legacy style decorator.
+            // TODO: Remove in next major.
+            if (typeof descriptor === 'string') {
+                const optNames = descriptor.split(`,`);
+
+                this.registerDefinition(prototype, command => {
+                    command.addOption({names: optNames, arity: 1, hidden});
+                });
+
+                this.registerTransformer(prototype, (state, command) => {
+                    for (const {name, value} of state.options) {
+                        if (optNames.includes(name)) {
+                            // @ts-ignore: The property is meant to have been defined by the child class
+                            command[propertyName] = command[propertyName] || [];
+                            // @ts-ignore: The property is meant to have been defined by the child class
+                            command[propertyName].push(value);
+                        }
+                    }
+                });
+            } else {
+                const SPECIAL_KEY = `__command_array`;
```

Comment:
Why not a symbol? This way we wouldn't have to worry about cleaning or hiding it